### DISCUSSION TITLE
For my project, I'd like to know how to [...] using Python

### DISCUSSION THREAD

**Discussion by felixchenier** [kineticstoolkit/kineticstoolkit#100] [2. Development: Ideas, new features, doc, etc.]
Hello all, and especially students.

I am slowly starting to develop tutorials for learning Python, NumPy, Matplotlib, Pandas and KTK for biomechanics, which will include many coding examples and exercises. Now would be a very good time to tell me:

For my project, I'd like to know how to [...] using Python

so I have ideas for real-life examples and exercises. If you're interested, just add your ideas to this discussion (describe a bit what you're doing, what kind of files you have, how is it organized, what do you want to calculate, etc.). Please don't be shy to post even if you think your needs are very easy or obvious; on the contrary, at this point, the simplest problems are very important to me.

Just to be clear, I won't answer your questions here. This is only a way for me to get ideas for tutorials that are focused on your needs.

Please share the word! 😃

**If you don't want to create a github account, you can also reply to this Tweet:** https://twitter.com/KineticsToolkit/status/1537897990316171264

**Comment by rogergallart** [comment]
I would like to see an example using the kinetics toolkit with IMUs. In my project I don't use optical markers but IMUs attached to each segment on the upper body (torso, upper arms, forearms, hands). Each IMU gives the orientation of the segment in form of a quaternion.

With the orientation (quaternions) and the anthropometry data it should be easy to calculate the coordinates of the joints and then use the toolkit functions to plot it, etc. A tutorial about this setup would be interesting.

Thanks

**Reply by felixchenier** [reply]
Thank you for your suggestion @rogergallart 
I find it very interesting and I think it requires a whole discussion of its own, therefore I started discussion #128 where we can better isolate what you propose and how we can get to it.
Thanks again!

**Comment by Yokonie** [comment]
Hello Félix Chénier,

For my project, I am looking at how lower limb joint kinetics (force and moment) and kinematics (joint angles) would change in response to exercise-induced fatigue during weightlifting exercises. 

With the joint kinematic data from the 3D motion capture system and ground reaction force data from the force plate, I would like to know how to get joint kinetic data (e.g., joint force and moment) with an inverse dynamics approach using Python as this is one of the things that I would like to look at in my project. 

Since from my understanding the estimation of segment mass, the position of the centre of mass, joint centres etc... would be required for calculating joint force and moment, I am not sure how easy or difficult this is going to be but if there is a tutorial for this using Python, I believe that many people would be interested.

Yoshi

**Reply by felixchenier** [reply]
@Yokonie 

Hello Yoshi

Thank you very much for your input! Inverse dynamics is a popular request, you're right that many people would be interested. I already developed bits of it for my own research, but the main challenge remains to get a nice, simple, practical yet powerful way to do it. That's not completely easy to figure 😁 

I started discussion #129 to at least officially start thinking about it. It could very well start with a "somewhat complicated" tutorial, then migrate to an extension to ktk, and finally be integrated right into ktk. I couldn't tell for now, but I assure you that I'll dedicate a space in my brain for it!

Don't hesitate to participate in #129 if you have ideas, as you progress into your project, and thanks again for your suggestion.

Félix
